OpenRA.Mods.Common/Traits/DamagedByTerrain.cs

```diff
@@ -92,8 +92,9 @@ void ITick.Tick(Actor self)
 			if (!self.IsInWorld)
 				return;
 
-			var t = self.World.Map.GetTerrainInfo(self.Location);
-			if (!Info.Terrain.Contains(t.Type))
+			var map = self.World.Map;
+			var cells = self.OccupiesSpace.OccupiedCells();
```

Comment:
At the risk of scope creeping things:
* We should try to cache this calculation so we only evaluate it if the actor moves
* We should apply the same fix to`GrantConditionOnTerrain`.

Would it be worth introducing a new `INotifyOccupiedCellsChanged` to simplify these?

Reply:
Caching was my initial idea as well, but the terrain can change underneath while not moving. Would have to look into somehow hooking into `CellEntryChanged`.